From 15d56d57c8d2c51fb64db4f2556c73492762ef9d Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Thu, 30 Oct 2008 15:00:31 +1100
Subject: [PATCH] xserver-kdrive-common: fix xserver-nodm to set HOME env var right

for a long time xserver-nodm has not set HOME correctly so all apps think HOME
is /. this fixes that.
---
 .../xserver-nodm-init/xserver-nodm                 |   17 ++++++++++-------
 1 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/packages/xserver-kdrive-common/xserver-nodm-init/xserver-nodm b/packages/xserver-kdrive-common/xserver-nodm-init/xserver-nodm
index 28ba65c..1f0fa47 100755
--- a/packages/xserver-kdrive-common/xserver-nodm-init/xserver-nodm
+++ b/packages/xserver-kdrive-common/xserver-nodm-init/xserver-nodm
@@ -19,13 +19,16 @@ done
 
 case "$1" in
   start)
-        . /etc/profile
-        [ -z $LOGNAME ] && export LOGNAME=root && export HOME=/home/root
-        [ -z $HOME ] && export HOME=/home/$LOGNAME
-
-
-        echo "Starting Xserver"
-        /etc/X11/Xserver &
+    # We don't want this script to block the rest of the boot process
+    if [ "$2" != "background" ]; then
+      $0 $1 background &
+    else
+       . /etc/profile
+       [ -z $LOGNAME ] && export LOGNAME=root && export HOME=/home/root
+       [ -z $HOME ] && export HOME=/home/$LOGNAME
+       echo "Starting Xserver"
+       /etc/X11/Xserver &
+    fi 
   ;;
 
   stop)
-- 
1.5.4.3

